cms/core/tests/test_db_router.py

```diff
@@ -55,5 +68,27 @@ def test_uses_correct_db_in_transaction(self):
 
     def test_search_uses_correct_db(self):
         """Check the read replica is used for search queries."""
-        with self.assertNumQueries(2, using=READ_REPLICA_DB_ALIAS), self.assertNumQueries(0, using=DEFAULT_DB_ALIAS):
+        with self.assertNumQueriesConnection(replica=1):
             list(HomePage.objects.search("Home"))
+
+    def test_deletes_from_default_when_read_from_replica(self):
+        user = User.objects.create_user("user", "email@example.com", "password")
+
+        self.assertEqual(user._state.db, DEFAULT_DB_ALIAS)  # pylint: disable=protected-access
+
+        # HACK: The replica connection can't see the user right now for some reason
+        user._state.db = READ_REPLICA_DB_ALIAS  # pylint: disable=protected-access
```

Comment:
note: I wish we didn't need to resort to hacks, but it does the job for the purpose of the test.